### DISCUSSION TITLE
How to use privateRuntimeConfig inside nuxt.config.js

### DISCUSSION THREAD

**Discussion by ghost** [nuxt/nuxt#10352] [Questions (Nuxt 2)]
I don’t want to compile my *local* env var `API_URL` (.env var) into my SPA.
Of course I want to have an extra env file on the server, where nuxt should take the `API_URL` env var.

I thought that I can solve this with privateRuntimeConfig, but I don’t know how to use it in my modules options.

Having
```js
privateRuntimeConfig: {
	apiUrl: process.env.API_URL,
},
```
How to integrate for example in auth config?
```js
auth: {
	strategies: {
		laravelSanctum: {
			provider: 'laravel/sanctum',
			// does this work online?
			// url: process.env.API_URL,
			url: this.$config.apiUrl, // ?????
			endpoints: {
				login: { url: '/api/login' },
				logout: { url: '/api/logout' },
			},
			user: { property: false }
		}
	}
},
```

How to protect my env vars from being exposed in the compiled nuxt app?

**Comment by pghorpade** [comment]
export default {
  privateRuntimeConfig: {
    apiSecret: process.env.API_SECRET
  }
}
this.$config.apiSecret is only available on server side
